### PR TITLE
fix(ci): harden and restore phase 1 gates

### PR DESCRIPTION
**Pull request by ss-o**
## Summary

- harden the Zsh compilation workflow against workflow-command injection by using NUL-delimited filename transport and positional parameters
- reconcile `next` development routing with `main` promotion guards, Renovate, and Dependabot behavior
- restore stable Go CI and bind releases to an annotated `vX.Y.Z` tag at the exact workflow event commit
- add workflow contract and mutation tests for the security and routing boundaries

## Why

Phase 1 addresses the workflow injection, branch-routing, and release-validation gaps tracked in #98 and #90. The underlying problems were unsafe filename interpolation into generated shell source, branch automation that was not reconciled with the repository's `next` development model, and release automation that did not bind the tag, checked-out commit, and workflow event SHA tightly enough.

## Impact

CI safely handles filenames containing shell metacharacters or newlines. Routine dependency updates target `next`. Promotion to `main` is restricted to the intended same-repository routes, including the authenticated Dependabot exception. Releases require an annotated semantic-version tag and run the uncached Go validation suite before creating one verified GitHub release.

No analyzer, parser, rule, command, or parser-backlog source is changed.

## Validation

- `go test -count=1 ./...`
- `go vet ./...`
- `go build -buildvcs=false ./...`
- `go mod tidy -diff`
- `gofmt -l .`
- actionlint 1.7.12
- Renovate 43.280.5 strict config validation
- Dependabot YAML structural validation
- signed commit, parent order, and final reviewed diff verification

## Known boundary

GitHub workflow event data can authenticate a same-repository `dependabot/*` pull request and its bot author, but cannot prove that a maintainer-retargeted bot PR is specifically a security update. The organization-level policy reconciliation is tracked separately in z-shell/.github#487.

## Tracking

- Addresses #90
- Phase 1 of #98
- Policy follow-up: z-shell/.github#487

### CODE REVIEW

File: .github/workflows/go-ci.yml [hunk 1]
```diff
@@ -21,7 +20,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: ⤵️ Check out code from GitHub
-        uses: actions/checkout@9c091bb21b7c1c1d1991bb908d89e4e9dddfe3e0 # v7.0.0
+        uses: actions/checkout@3d3c42e5aac5ba805825da76410c181273ba90b1 # v7.0.1
```
[the remark this answers is not in the document]
Comment (ss-o) [reply .github/workflows/go-ci.yml:23]: Fixed in `bccb6fb`: `go-ci.yml` now sets `persist-credentials: false` on its checkout step. `TestPullRequestValidationCheckoutsDoNotPersistCredentials` covers both Go CI and docs generation. Fresh local Go 1.25.12 tests, vet, build, module tidiness, gofmt, and actionlint pass, and the PR checks are green.
